load-balanced-emr-dynamodb-hadoop/src/main/java/org/apache/hadoop/dynamodb/read/LoadBalancedDynamoDBRecordReader.java

```diff
@@ -38,26 +39,23 @@
  * convertDynamoDBItemToValue with a conversion from the DynamoDB item to the desired value type. As
  * an alternative, you can also use DefaultDynamoDBRecordReader, and do the conversion to a
  * DynamoDBItemWritable inside of the reducer.
- *
- * @param <K> The type of Key that will be passed to the Mapper
- * @param <V> The type of Value that will be passed to the Mapper
```

Comment:
I understand you are simplifying, but I don't get how only Text can be the key
(I didn't check DynamoDBItemWritable but I assume it's an interface to most data types value can have in Dynamo)

Reply:
Hadoop requires a key, but here they always use an [empty `Text` instance](https://github.com/awslabs/emr-dynamodb-connector/blob/ef1385cc00f11664046e6cb388f81fb84b776964/emr-dynamodb-hadoop/src/main/java/org/apache/hadoop/dynamodb/read/DefaultDynamoDBRecordReader.java#L28-L30) as a key. So, it is not really used. `DynamoDBItemWritable` is a class that only contains a `Map<String, AttributeValue>` (where `AttributeValue` comes from the AWS Java SDK and models a column in a DynamoDB row).